server/services/analyticsService.ts

```diff
@@ -480,6 +480,72 @@ export default class AnalyticsService {
       populationIsTotal: true,
     }
   }
+
+  async getIncentiveLevelTrendsByCharacteristic(
+    prison: string,
+    protectedCharacteristic: ProtectedCharacteristic,
+    characteristicGroup: string
+  ): Promise<TrendsReport> {
+    const columnsToStitch = ['prison', 'year_month_str', 'snapshots', 'offenders', 'incentive', protectedCharacteristic]
+    type StitchedRow = [string, string, number, number, string, string]
+
+    const { stitchedTable, date: lastUpdated } = await this.cache.getStitchedTable<TrendsTable, StitchedRow>(
+      this,
+      TableType.trends,
+      columnsToStitch
+    )
+
+    const columnSet: Set<string> = new Set()
+    const filteredTables = stitchedTable.filter(
+      ([somePrison, _yearAndMonth, _snapshots, _offenders, incentive, someCharacteristicGroup]) => {
+        const include =
+          // filter only selected prison
+          somePrison === prison &&
+          someCharacteristicGroup.trim() === characteristicGroup &&
+          // it's possible for incentive level to be null
+          incentive
+        if (include) {
+          columnSet.add(incentive)
+        }
+        return include
+      }
+    )
+    if (filteredTables.length === 0) {
+      throw new AnalyticsError(AnalyticsErrorType.EmptyTable, 'Filtered trends report for incentive levels has no rows')
```

Comment:
i'd include the PC here so that it can be distinguished from the not-filtered-by-PC error message

Reply:
Just PC or should we spell out the value for PC/PC group as well you think?

e.g.
a. `Filtered PC trends report for incentive levels has no rows`
b. `Filtered PC trends report for incentive levels for ${protectedCharacteristic}/${characteristicGroup} has no rows`

I think option a. may be enough (I'll made this tweak) but if you think we should be even more explicit let me know.

Reply:
https://github.com/ministryofjustice/hmpps-incentives-ui/pull/190/commits/3519f1ceb0a3030f552c0f34210cc0320790e782